### ISSUE TITLE
Overridable deep equal and object filter

### ISSUE THREAD

**Issue by mikaelbr** [omniscientjs/omniscient#34]
@Dashed wrote in #33 :

> I want to bring this to your attention: https://github.com/dustingetz/react-cursor/blob/master/src/ImmutableOptimizations.js

> Unsure if it's any useful, but they have configurable keys as a whitelist for reference check (===) instead of deep equal. Not only that, but they also have ignore keys which may be added to isNotIgnorable ruleset.

>Then as an idea:

> ```js
> var shouldUpdate = require('omniscient/shouldupdate');

> shouldupdate; // default shouldComponentUpdate

>shouldComponentUpdate({ deepEqual: _.isEqual, objectFilter: _.filter, ... });
> ````
> Can we be able to change filterKeyValue and deepEqual? This is probably useful if we want to use our favourite implementations of object-filter or deep-equal; usually using libraries suitable for particular environment (e.g. browser/node.js).

**Comment by Dashed** [comment]
@mikaelbr Now that individual lodash functions are published to npm, do you think it'll be desirable to use them over the current libs?

https://www.npmjs.com/package/lodash.filter
https://www.npmjs.com/package/lodash.isequal

**Comment by mikaelbr** [comment]
I think this is sufficient: https://github.com/omniscientjs/omniscient/blob/6a88e50a65b475d901739c52e66fa407a2df49e9/shouldupdate.js#L16 ?

Is there really need to be able to override `_.filter` when you can override props and state checks?

**Comment by Dashed** [comment]
I glanced at `shouldUpdate.js` before and after the changes you've made. 

- https://github.com/omniscientjs/omniscient/blob/68795b5aa766a3bb99b37f7836d25cb72fadbbca/shouldupdate.js
- https://github.com/omniscientjs/omniscient/blob/ff6147b3c59a31b7dae28377c5fec4c4777de6b2/shouldupdate.js

Although it's possible to override the lodash pick/isEqual lib (via `methods`), it may require the user to know how it's used internally in `shouldComponentUpdate` (e.g. same function signature).

Plus, lodash functions are already very stable, and it's unlikely to be replaced by something else for a while. Initially, I wanted to override the substack's `node-deep-equal` is because of some bugs I've encountered with it. lodash's isEqual doesn't suffer those same bugs :smile: . Now, I don't have any need to override the pick/isEqual libs.

Instead, the end-user would be more interested in overriding `isEqualProps` and `isEqualState`.

@mikaelbr So, I think overriding these libs is no longer necessary. Feel free to close :)
